src/main/java/steps/CartPageSteps.java

```diff
@@ -16,4 +23,58 @@ public static void goToCheckout() {
         cartPage = new CartPage();
         cartPage.clickOnFinalCheckoutButton();
     }
-}
+
+    public static List<ProductModel> getProductsFromShoppingCartTable() {
+        cartPage = new CartPage();
+        return cartPage.shoppingCartTable.getProducts();
+    }
+
+    public static List<CheckoutProductModel> getCheckoutProductsFromShoppingCartTable() {
+        cartPage = new CartPage();
+        return cartPage.shoppingCartTable.getCheckoutProducts();
+    }
+
+    public static void removeProductFromShoppingCartTable(CheckoutProductModel checkoutProduct) {
+        cartPage = new CartPage();
+        var shoppingCartTableRow = new ShoppingCartTableRow(cartPage.shoppingCartTable
+                .getCheckoutProducts()
+                .indexOf(checkoutProduct) + 1);
+        shoppingCartTableRow.removeProduct();
+    }
+
+    public static void changeQuantityOfProduct(ProductModel product, int quantity) {
+        cartPage = new CartPage();
+        var shoppingCartTableRow = new ShoppingCartTableRow(cartPage.shoppingCartTable
+                .getProducts()
+                .indexOf(product) + 1);
+        shoppingCartTableRow.setQuantity(quantity);
+    }
+
+    public static void checkIfCartIsEmpty() {
+        cartPage = new CartPage();
+        Assert.assertTrue(cartPage.shoppingCartTable.isCartEmpty());
+    }
+
+    public static void checkIfProductAppearsOnce(ProductModel product, List<CheckoutProductModel> checkoutProducts) {
+        var count = checkoutProducts.stream()
+                .filter(p -> p.getProductModel().equals(product))
+                .count();
+        Assert.assertEquals(1, count);
+    }
+
+    public static void checkIfTotalPriceIsCorrect(List<ProductModel> products, List<CheckoutProductModel> checkoutProducts) {
+        var unitPrices = products.stream()
+                .map(ProductModel::getPrice)
+                .toList();
```

Comment:
you should be able to use something like that:
.collect(Collectors.summingInt(Integer::intValue));
or similiar depending on the type on value that you use:)